code/game/gamemodes/vampire/vampire.dm

```diff
@@ -295,69 +313,89 @@ You are weak to holy things and starlight. Don't go into space and avoid the Cha
 		H.LAssailant = null
 	else
 		H.LAssailant = owner
-	while(do_mob(owner, H, 50))
+	while(do_mob(owner, H, suck_rate))
 		if(!(owner.mind in SSticker.mode.vampires))
 			to_chat(owner, "<span class='userdanger'>Your fangs have disappeared!</span>")
 			return
-		old_bloodtotal = bloodtotal
-		old_bloodusable = bloodusable
-		if(H.stat < DEAD)
-			if(H.ckey || H.player_ghosted) //Requires ckey regardless if monkey or humanoid, or the body has been ghosted before it died
-				blood = min(20, H.blood_volume)	// if they have less than 20 blood, give them the remnant else they get 20 blood
-				bloodtotal += blood / 2	//divide by 2 to counted the double suction since removing cloneloss -Melandor0
-				bloodusable += blood / 2
-		else
-			if(H.ckey || H.player_ghosted)
-				blood = min(5, H.blood_volume)	// The dead only give 5 blood
-				bloodtotal += blood
-		if(old_bloodtotal != bloodtotal)
-			if(H.ckey || H.player_ghosted) // Requires ckey regardless if monkey or human, and has not ghosted, otherwise no power
-				to_chat(owner, "<span class='notice'><b>You have accumulated [bloodtotal] [bloodtotal > 1 ? "units" : "unit"] of blood[bloodusable != old_bloodusable ? ", and have [bloodusable] left to use" : ""].</b></span>")
-		check_vampire_upgrade()
-		H.blood_volume = max(H.blood_volume - 25, 0)
-		//Blood level warnings (Code 'borrowed' from Fulp)
-		if(H.blood_volume)
-			if(H.blood_volume <= BLOOD_VOLUME_BAD && blood_volume_warning > BLOOD_VOLUME_BAD)
-				to_chat(owner, "<span class='danger'>Your victim's blood volume is dangerously low.</span>")
-			else if(H.blood_volume <= BLOOD_VOLUME_OKAY && blood_volume_warning > BLOOD_VOLUME_OKAY)
-				to_chat(owner, "<span class='warning'>Your victim's blood is at an unsafe level.</span>")
-			blood_volume_warning = H.blood_volume //Set to blood volume, so that you only get the message once
-		else
-			to_chat(owner, "<span class='warning'>You have bled your victim dry!</span>")
-			break
+		if(unique_suck_id in drained_humans)
+			if(drained_humans[unique_suck_id] >= BLOOD_DRAIN_LIMIT)
+				to_chat(owner, "<span class='warning'>You have drained most of the life force from [H]'s blood, and you will get no useable blood from them!</span>")
+				H.blood_volume = max(H.blood_volume - 25, 0)
+			else
+				if(H.stat < DEAD)
+					if(H.ckey || H.player_ghosted) //Requires ckey regardless if monkey or humanoid, or the body has been ghosted before it died
+						blood = min(20, H.blood_volume)
+						bloodtotal += blood * BLOOD_GAINED_MODIFIER
+						bloodusable += blood * BLOOD_GAINED_MODIFIER
+						drained_humans[unique_suck_id] += blood * BLOOD_GAINED_MODIFIER
+						check_vampire_upgrade()
+						to_chat(owner, "<span class='notice'><b>You have accumulated [bloodtotal] [bloodtotal > 1 ? "units" : "unit"] of blood, and have [bloodusable] left to use.</b></span>")
```

Comment:
Can you even get just 1 unit of blood?

Reply:
nope!

Reply:
Byond has a built-in `\s` macro which could work for this too.
http://www.byond.com/docs/ref/#/DM/text/macros
```dm
to_chat(owner, "<span class='notice'><b>You have accumulated [bloodtotal] unit\s of blood, and have [bloodusable] left to use.</b></span>")
```